artemis-core-client/src/main/java/org/apache/activemq/artemis/core/client/impl/ServerLocatorImpl.java

```diff
@@ -737,12 +739,13 @@ public ClientSessionFactory createSessionFactory() throws ActiveMQException {
       // how the sendSubscription happens.
       // in case this ever changes.
       if (topology != null && !factory.waitForTopology(config.callTimeout, TimeUnit.MILLISECONDS)) {
+         synchronized (factories) {
+            factories.remove(factory);
```

Comment:
I think you should create a private method removeFactory(factory)

and that method should do the synchronized(factories)


just for readability?

Reply:
I reused `factoryClosed`